modules/imap/functions.php

```diff
@@ -579,26 +579,56 @@ function format_msg_part_section($struct, $output_mod, $part, $dl_link, $at_link
     if ($mobile) {
         $simple_view = true;
     }
-    foreach ($struct as $id => $vals) {
-        if (is_array($vals) && isset($vals['type'])) {
-            $row = format_msg_part_row($id, $vals, $output_mod, $level, $part, $dl_link, $at_link, $use_icons, $simple_view, $mobile);
-            if (!$row) {
-                $level--;
-            }
-            $res .= $row;
-            if (isset($vals['subs'])) {
-                $res .= format_msg_part_section($vals['subs'], $output_mod, $part, $dl_link, $at_link, ($level + 1));
+
+    if(!$simple_view){
+        foreach ($struct as $id => $vals) {
+            if (is_array($vals) && isset($vals['type'])) {
+                $row = format_msg_part_row($id, $vals, $output_mod, $level, $part, $dl_link, $use_icons, $simple_view, $mobile);
+                if (!$row) {
+                    $level--;
+                }
+                $res .= $row;
+                if (isset($vals['subs'])) {
+                    $res .= format_msg_part_section($vals['subs'], $output_mod, $part, $dl_link, ($level + 1));
+                }
             }
-        }
-        else {
-            if (is_array($vals) && count($vals) == 1 && isset($vals['subs'])) {
-                $res .= format_msg_part_section($vals['subs'], $output_mod, $part, $dl_link, $at_link, $level);
+            else {
+                if (is_array($vals) && count($vals) == 1 && isset($vals['subs'])) {
+                    $res .= format_msg_part_section($vals['subs'], $output_mod, $part, $dl_link, $level);
+                }
             }
         }
+    }else{
+        $res = format_attachment($struct,  $output_mod, $part, $dl_link);
     }
     return $res;
 }}
 
+function format_attachment ($struct,  $output_mod, $part, $dl_args) {
+    $res = '';
+
+    foreach ($struct as $id => $vals) {
+        if(is_array($vals) && $vals['type'] != 'multipart' && isset($vals['file_attributes']) && !empty($vals['file_attributes'])) {
```

Comment:
isset($vals['type']) - same like the block above in not simple view if statement - I am afraid, otherwise, $vals can be an array but missing type key will issue a warning.

Reply:
Done